src/Storage/Node.cs

```diff
@@ -2677,6 +2677,124 @@ private void MoveTo(Node target, long sourceTimestamp, long targetTimestamp)
             }
         }
 
+        /// <summary>
+        /// Modes the Node instance to another loacation. The new location is a Node instance which will be parent node.
+        /// </summary>
+        public virtual Node MoveToAndGetMoved(Node target)

```

Comment:
I think these new methods should not be _virtual_, we do not want to extend the api that way, just add more base functionality.
Also, please do not copy full method source code, make old methods that do not return results call the new methods to avoid code duplucation.